### ISSUE TITLE
Unknown CPU family 'i86pc' on illumos

### ISSUE THREAD

**Issue by Mno-hime** [mesonbuild/meson#4146]
I build `meson` for OpenIndiana (illumos kernel, OpenSolaris fork) and test suite in many places fails like this:
```
The Meson build system
Version: 0.47.2
Source dir: /export/home/newman/ws/oi-userland/components/developer/meson/meson-0.47.2/test cases/common/46 library chain
Build dir: /tmp/tmplt9q5vqn
Build type: native build
Project name: libchain
Project version: undefined
Native C compiler: /usr/gcc/6/bin/gcc (gcc 6.4.0 "gcc (OpenIndiana 6.4.0-OI-3) 6.4.0")
WARNING: Unknown CPU family 'i86pc', please report this at https://github.com/mesonbuild/meson/issues/new with theoutput of `uname -a` and `cat /proc/cpuinfo`
Build machine cpu family: i86pc
Build machine cpu: i86pc
Build targets in project: 4
Found ninja-1.7.2 at /usr/bin/ninja
```
`uname -a` here is: `SunOS build-userland 5.11 illumos-b075503f22 i86pc i386 i86pc`.

We don't have `/proc/cpuinfo`, attaching output of [`cpuid`](https://github.com/mesonbuild/meson/files/2365674/cpuid.txt) instead. What additional info you might need?

**Comment by Mno-hime** [comment]
Also have errors of this sort:
```
The Meson build system
Version: 0.47.2
Source dir: /export/home/newman/ws/oi-userland/components/developer/meson/meson-0.47.2/test cases/common/140 get define
Build dir: /tmp/tmpb798zkn9
Build type: native build
Project name: get define
Project version: undefined
Appending CFLAGS from environment: '-DMESON_FAIL_VALUE=cflags-read'
Appending CPPFLAGS from environment: '-DMESON_TEST_DEFINE_VALUE="spaces and fun@$&()-=_+{}[]:;>?,./~`"'
Native C compiler: /usr/gcc/6/bin/gcc (gcc 6.4.0 "gcc (OpenIndiana 6.4.0-OI-3) 6.4.0")
Appending CPPFLAGS from environment: '-DMESON_TEST_DEFINE_VALUE="spaces and fun@$&()-=_+{}[]:;>?,./~`"'
Native C++ compiler: c++ (gcc 6.4.0 "c++ (OpenIndiana 6.4.0-OI-3) 6.4.0")
Build machine cpu family: i86pc
Build machine cpu: i86pc

meson.build:36:4: ERROR:  Problem encountered: Please report a bug and help us improve support for this platform
```

**Comment by jpakkane** [comment]
Is this 32 bit or 64? Or does it print the same for both?

**Comment by alanc** [comment]
It prints the same for both.  When I was working on porting meson to Solaris last year, I solved this with https://github.com/alanc/meson/commit/1a0f803109a44301a1ae7d86e8fe283ea37ac595 (but I had to put that work aside and never completed it, due to changes in my responsibilities at work).

**Comment by Mno-hime** [comment]
Kernel is 64-bit, can run both 32-bit and 64-bit apps. Output is the same for both. Bitness can be determined by `isarch`:
```
$ isainfo 
amd64 i386
$ isainfo -b
64
```
Don't know if Python's platform module can determine that somehow.

**Comment by jpakkane** [comment]
Probably the best solution is to add a check that calls the `isarch` program if it encounters `i86pc`. Pull requests are welcome, I don't think anyone on the dev team even has a Solaris machine to develop and verify this on.

**Comment by chincheta0815** [comment]
@Mno-hime Did you get any further with this issue? While trying to update tracker I ran into a missing mesonbuild and got that same issue. I can get rid of the error message but then meson complains about the as-needed option of the linker. This seems due to the specialties of OpenIndiana's linker (ld vs. gld)

**Comment by dcbaker** [comment]
I took alanc's branch and got a bit further with it, but I ran into issues with needing to tease apart meson's assumption that gcc/clang will always have gnu-ld (@jon-turney ran into the same issue for something else I think?), and never got finished with that.

**Comment by dcbaker** [comment]
for the linker issue see: #4784

**Comment by chincheta0815** [comment]
oh cool. I also workd on that. I got a whole bunch of patches. I still do not have all tests running.
My assumption was that there is solaris ld as a default.

https://github.com/chincheta0815/oi-userland/tree/meson/components/developer/meson/patches

**Comment by vedranmiletic** [comment]
> oh cool. I also workd on that. I got a whole bunch of patches. I still do not have all tests running.
> My assumption was that there is solaris ld as a default.
> 
> https://github.com/chincheta0815/oi-userland/tree/meson/components/developer/meson/patches

This link is now dead. Any progress? I still get this with Meson 0.51 on OpenIndiana Hipster 2019.04.

**Comment by jpakkane** [comment]
It should be in master: https://github.com/mesonbuild/meson/commit/a35844c08d8d30e21920c4881ce2f859961c5a0d

**Comment by vedranmiletic** [comment]
> It should be in master: [a35844c](https://github.com/mesonbuild/meson/commit/a35844c08d8d30e21920c4881ce2f859961c5a0d)

2017? So this is part of 0.51? Because I just got the message on OpenIndiana Hipster 2019.04.

**Comment by jpakkane** [comment]
No, it was merged after 0.51.0 was released. It will be in 0.52.0.

**Comment by dcbaker** [comment]
I'm going to close this as fixed.
